src/Microsoft.OData.Core/JsonLight/ODataJsonLightServiceDocumentSerializer.cs

```diff
@@ -132,5 +133,131 @@ private void WriteServiceDocumentElement(ODataServiceDocumentElement serviceDocu
+                        }
+                    }
+
+                    if (serviceDocument.Singletons != null)
+                    {
+                        foreach (ODataSingletonInfo singletonInfo in serviceDocument.Singletons)
+                        {
+                            await this.WriteServiceDocumentElementAsync(singletonInfo, JsonLightConstants.ServiceDocumentSingletonKindName)
+                                .ConfigureAwait(false);
+                        }
+                    }
+
+                    HashSet<string> functionImportsWritten = new HashSet<string>(StringComparer.Ordinal);
+
+                    if (serviceDocument.FunctionImports != null)
+                    {
+                        foreach (ODataFunctionImportInfo functionImportInfo in serviceDocument.FunctionImports)
+                        {
+                            if (functionImportInfo == null)
+                            {
+                                throw new ODataException(Strings.ValidationUtils_WorkspaceResourceMustNotContainNullItem);
+                            }
+
+                            if (!functionImportsWritten.Contains(functionImportInfo.Name))
+                            {
+                                functionImportsWritten.Add(functionImportInfo.Name);
+                                await this.WriteServiceDocumentElementAsync(functionImportInfo, JsonLightConstants.ServiceDocumentFunctionImportKindName)
+                                    .ConfigureAwait(false);
+                            }
+                        }
+                    }
+
+                    // "]"
+                    await this.AsynchronousJsonWriter.EndArrayScopeAsync()
+                        .ConfigureAwait(false);
+
+                    // "}"
+                    await this.AsynchronousJsonWriter.EndObjectScopeAsync()
+                        .ConfigureAwait(false);
+                });
+        }
+
+        /// <summary>
+        /// Asynchronously writes a element (EntitySet, Singleton or FunctionImport) in service document.
+        /// </summary>
+        /// <param name="serviceDocumentElement">The element in service document to write.</param>
+        /// <param name="kind">Kind of the service document element, optional for entitysets must for FunctionImport and Singleton.</param>
+        private async Task WriteServiceDocumentElementAsync(ODataServiceDocumentElement serviceDocumentElement, string kind)
+        {
+            // validate that the resource has a non-null url.
+            ValidationUtils.ValidateServiceDocumentElement(serviceDocumentElement, ODataFormat.Json);
+
+            // "{"
+            await this.AsynchronousJsonWriter.StartObjectScopeAsync()
+                .ConfigureAwait(false);
+
+            // "name": ...
+            await this.AsynchronousJsonWriter.WriteNameAsync(JsonLightConstants.ODataServiceDocumentElementName)
+                .ConfigureAwait(false);

```

Comment:
Is the separator `:` included here?

Reply:
@habbes All variants of `WriteName`/`WriteNameAsync` always write the separator